flow/layers/clip_rrect_layer.cc

```diff
@@ -37,8 +37,7 @@ void ClipRRectLayer::Paint(PaintContext& context) {
   TRACE_EVENT0("flutter", "ClipRRectLayer::Paint");
   FTL_DCHECK(!needs_system_composite());
 
-  SkAutoCanvasRestore save(&context.canvas, false);
-  context.canvas.saveLayer(&paint_bounds(), nullptr);
+  Layer::AutoSaveLayer save(context, paint_bounds(), nullptr);
```

Comment:
In the previous version of this code, we were doing a `SkCanvas::save (via SkAutoCanvasRestore)` -> `SkCanvas::saveLayer` -> PaintChildren -> `SkCanvas::restore (via SkAutoCanvasRestore)`. In this version, we no longer have the `SkCanvas::save`. Is this fine? I am not sure why we did the SkCanvas::save in the first place.